telemetry/prometheus/src/main/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeter.scala

```diff
@@ -44,14 +44,19 @@ class PrometheusTelemeter(metrics: MetricsTree, private[prometheus] val handlerP
   private[this] val labelValDisallowedChars = """(\\|\"|\n)""".r
   private[this] def escapeLabelVal(key: String) = labelValDisallowedChars.replaceAllIn(key, """\\\\""")
 
-  private[this] def formatLabels(labels: Seq[(String, String)]): String =
+  private[this] def formatLabels(labels: Seq[(String, String)], sb: StringBuilder): Unit =
     if (labels.nonEmpty) {
-      labels.map {
-        case (k, v) =>
-          s"""${escapeLabelKey(k)}="${escapeLabelVal(v)}""""
-      }.mkString("{", ", ", "}")
-    } else {
-      ""
+      val lastIndex = labels.size - 1
+      sb.append("{")
+      for (((k, v), i) <- labels.zipWithIndex) {
+        sb.append(k)
+        sb.append("=\"")
+        sb.append(v)
```

Comment:
If I understand correctly, this change means that all key/vals in `labels0` now skip `escapeLabelKey` and `escapeLabelVal`. Do we need to take another pass over `labels0` to do escaping?

Reply:
that's correct but all the label names are defined as literals in this file and none of them contain any characters that need escaping.